src/Data/List/Properties.agda

```diff
@@ -122,6 +122,15 @@ map-injective finj {x ∷ xs} {y ∷ ys} eq =
 ------------------------------------------------------------------------
 -- mapMaybe
 
+module _ {f g : A → Maybe B} where
+
+  mapMaybe-cong : f ≗ g → mapMaybe f ≗ mapMaybe g
+  mapMaybe-cong f≗g [] = refl
+  mapMaybe-cong f≗g (x ∷ xs) rewrite f≗g x
+    with ih ← mapMaybe-cong f≗g xs | g x
```

Comment:
Why do we need the ih in a with? Why can't we just write `cong (y ∷_) (mapMaybe-cong f≗g xs)`? Similarly for proofs below?

Reply:
I was mirroring the style of the existing proofs. I figured they put ih in a with to avoid repetiton in both branches, so I did the same here. I can change that if you prefer

Reply:
See also #2123 , second checkbox, on uses of 'irrefutable' `with` (when a `let` mucks up the flow of pattern-matching; LHS `let` will be a boon here)